src/GRA.Data/Repository/EmailReminderRepository.cs

```diff
@@ -23,5 +27,30 @@ public async Task<bool> ExistsEmailSourceAsync(string emailAddress, string signU
                 .SingleOrDefaultAsync();
             return lookup != null;
         }
+
+        public IEnumerable GetDistinctSignUpSourceWithCount()
+        {
+            return DbSet
+                .GroupBy(_ => _.SignUpSource)
+                .Select(_ => new
+                {
+                    DisplayText = _.Key + " (" + _.Distinct().Count().ToString() + ")",
```

Comment:
Repository should return database items and the business logic layer should be altering those for return to the front end.